src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,173 @@
+
+        if (!timeStr.isBlank()) {
+            try {
+                time = timeFormatter.parse(timeStr, LocalTime::from);
+            } catch (DateTimeParseException e) {
+                throw new DukeException("Unable to parse time.\n \n"
+                        + "Please input your time in one of the following formats:\n"
+                        + "1:19\n" + "1:19 AM");
+            }
+        }
+
+        return time;
+    }
+
+    /**
+     * Parses the command given and returns a Command object for execution.
+     *
+     * @param fullCommand A string containing the full command.
+     * @return a Command object representing the command.
+     * @throws DukeException if the command cannot be parsed.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] commands = fullCommand.split("\\s");
+        String command = commands[0];
+        String args = commands.length > 1 ? String.join(" ", Arrays.copyOfRange(commands, 1, commands.length)) : "";
+
+        switch (command) {
+        case "bye":
+            if (!args.isBlank()) {
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            }
+
+            return new ExitCommand();
+        case "list":
+            if (!args.isBlank()) {
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            }
+
+            return new ListCommand();
+        case "due":
+            if (args.isBlank()) {
+                throw new DukeException("Date required for the due command.");
+            }
+
+            return new DueCommand(Parser.parseDate(args));
+        case "done":
+            if (args.isBlank()) {
+                throw new DukeException("Task number required for the done command.");
+            } else if (!args.chars().allMatch(Character::isDigit)) {
+                throw new DukeException("Only positive integers allowed for the done command.");
+            }
+
+            return new DoneCommand(Integer.parseInt(args));
+        case "delete":
+            if (args.isBlank()) {
+                throw new DukeException("Task number required for the delete command.");
+            } else if (!args.chars().allMatch(Character::isDigit)) {
+                throw new DukeException("Only positive integers allowed for the delete command.");
+            }
+
+            return new DeleteCommand(Integer.parseInt(args));
+        case "find":
+            if (args.isBlank()) {
+                throw new DukeException("Keyword cannot be blank.");
+            }
+
+            return new FindCommand(args);
+        case "todo":
+            return new AddCommand(TaskType.TODO, args);
+        case "deadline": {
+            String[] deadlineArgs = args.split("\\s/by\\s");
+            String description = deadlineArgs[0];
+            String by = deadlineArgs.length > 1
```

Comment:
Maybe you want to rename the variable "by" as something like "deadlineTime"/"deadlineDate"? It could be clearer and easier for a person who doesn't know Duke Project to read 😄